### PR TITLE
Show other eligibility criteria to all teachers regardless of school eligibility

### PR DESCRIPTION
**Pull request by bencodeorg**
[PLC-742](https://codedotorg.atlassian.net/browse/PLC-742)

- Makes discount code application criteria visible to all applicants, instead of only to those whose schools are eligible. Implemented to make life easier for internal folks who are corresponding with teachers directly and want teachers to be able to access information about their status.
- Adds the school eligibility check as one of the check/x steps in the "Eligibility Requirements" list -- redundant, but maybe more clear than not including it.

Before:

image

After (the "Eligibility Requirements" checklist is already visible to teachers who met the school bar):

image

## Testing story

Added tests to:
- confirm that one of the eligibility checks (the Unit 6 check) renders if a teacher's school does not meet the school eligibility bar. Tried this test without the new code (failed), and with the code changes (passed).
- confirm that the "Submit" button does not appear if a teacher has met the first two "Other" eligibility criteria but not the school eligibility criteria.

## Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/kits/maker/ui/EligibilityChecklist.jsx [hunk 1]
```diff
@@ -95,6 +95,14 @@ export default class EligibilityChecklist extends React.Component {
     });
   };
 
+  schoolHighNeedsStatusMessage = schoolHighNeedsEligible => {
+    return schoolHighNeedsEligible === true ? (
```
Comment (breville) [inline apps/src/lib/kits/maker/ui/EligibilityChecklist.jsx:99]: Curious: Does it need to be explicitly `true` or would truthy be enough?

**Review by breville — APPROVED** [review]

File: apps/src/lib/kits/maker/ui/EligibilityChecklist.jsx (hunk 1 above)
Comment (bencodeorg) [reply apps/src/lib/kits/maker/ui/EligibilityChecklist.jsx:99]: mmm truthy would probably be enough? The argument should always either be `true` or `false` in the one place that this method is called, so I figured it was better to be explicit about it. Not sure if that's the right call, though.

**Comment by molly-moen** [comment]
Will it be confusing if the teacher sees only green check marks (since there's no red x for school not being eligible) but they aren't able to get the code?

**Comment by agealy** [comment]
> Will it be confusing if the teacher sees only green check marks (since there's no red x for school not being eligible) but they aren't able to get the code?

Good callout. Adding a "Your school is eligible for subsidy" below the school name with the same check/x formatting could help

**Comment by bencodeorg** [comment]
@agealy @molly-moen your feedback triggered a realization that my initial attempt didn't account for the case where a teacher fails the school check, but passes both of the first two "Other" checks -- they would have previously seen a "Submit" button that would have given them an error. I've fixed that in my most recent commits.

It'd be a decent amount of work to implement the same check/x type pattern that exists for the "Other Eligibility Criteria" section in the "School Eligibility" section -- given the low volume this page sees, I'm inclined not to do that work. I did bold the message saying "your school is not eligible" as a low effort approach (see screenshot)?

**Review by molly-moen — APPROVED** [review]
I think bolding is fine since the page is minimally used. One other option (if it's easy) would be to add "Your school is eligible" above the other eligibility requirements in the section that already contains the x's/checks, so that section contains the full list of eligibility requirements.

**Comment by agealy** [comment]
@bencodeorg I agree, it's the right intuition to keep this minimal. Bold formatting is sufficient, and Molly's suggestion would help too if it's easy, but is not necessary

**Comment by bencodeorg** [comment]
@agealy added a big checkmark for whether the teacher passed the school validation -- duplicates information, but hopefully clear enough?

**Comment by agealy** [comment]
Perfect!

**Review by molly-moen — APPROVED** [review]

**Pull request merged**